### PR TITLE
fozzie-components@v1.4.0 – Adding DangerJS checks to repo

### PR DESCRIPTION
**Pull request by ashleynolan**
### Added
- DangerJS Checks (via CircleCI)
- Pull Request Template
- `CODEOWNERS` file

### Changed
- #trivial changes to package scripts changing `prepare` hooks to `prepublishOnly` (so they only run on `npm publish` not on `npm install`)

### CODE REVIEW

File: packages/f-header/package.json [hunk 1]
```diff
@@ -41,7 +41,8 @@
     "lodash-es": "4.17.15"
   },
   "peerDependencies": {
-    "@justeat/f-trak": "0.6.0"
+    "@justeat/f-trak": "0.6.0",
+    "@justeat/browserslist-config-fozzie": ">=1.1.1"
```
Comment (DamianMullins) [inline packages/f-header/package.json]: This is already in the root package, do you need it here as well?

Comment (DamianMullins) [reply packages/f-header/package.json]: Just noticed it's a peer dep. Is it needed in f-footer as well?

File: dangerfile.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    // Get an array of files in the root monorepo that have changed
+    const modifiedRootFiles = modifiedFiles.filter(filepath => !filepath.startsWith('packages/') && filepath !== 'yarn.lock');
+
+    // Get an array of packages that have changed, then make that array unique
+    const modifiedPackages = modifiedFiles.filter(filepath => filepath.startsWith('packages/'))
+        .map(filepath => filepath.split('/')[1]);
+    const uniqueModifiedPackages = new Set(modifiedPackages);
+
+    // Fail if the title of the PR isn't in the format of a version i.e. vX.X.X (such as v1.4.0)
```
Comment (xander-marjoram) [inline dangerfile.js]: Example in the comment (and the `fail`) should be something like `{package-name}@vX.X.X` to match the regex.

Comment (ashleynolan) [reply dangerfile.js]: Good catch – updated

File: packages/f-header/package.json (hunk 1 above)
Comment (ashleynolan) [reply packages/f-header/package.json]: I've removed this for now as not sure it's needed for this PR (and may need to bump package version separately). Will check if it's needed when I bump header/footer to v2 after this gets merged…

**Comment by xander-marjoram** [comment]
@ashleynolan Can you bump the package versions and add a changelog entry for each package? :trollface:

**Review by xander-marjoram — DISMISSED** [review]

**Review by xander-marjoram — APPROVED** [review]

**Pull request merged**